fetch.bs

```diff
@@ -3763,11 +3921,41 @@ steps:
 
  <li><p>Wait for either <var>response</var>'s <a for=response>body</a> to be null, or
  <var>response</var>'s <a for=response>body</a>'s <a for=body>stream</a> to be
- <a for=ReadableStream>closed</a> or <a for=ReadableStream>errored</a>, and then set
- <var>request</var>'s <a>done flag</a>.
- <!-- This is really bad and needs to be handled differently at some point. -->
+ <a for=ReadableStream>closed</a> or <a for=ReadableStream>errored</a>, and perform the following steps:
+ <ol>
+  <li><p>Set <var>request</var>'s <a>done flag</a>.
+   <!-- This is really bad and needs to be handled differently at some point. -->
+ </ol>
 </ol>
 
+<p>To <dfn>handle response end</dfn>, given <a for=/>fetch params</a> <var>fetchParams</var> and
+<a for=/>response</a> <var>response</var>, perform the following steps:
+<ol>
+  <li>Set <var>fetchParams</var>'s <a for="fetch params">timing info</a>'s
+  <a for="fetch timing info">response end time</a> to the <a for=/>unsafe shared current time</a>.
+  <li>Let <var>request</var> be <var>fetchParams</var>'s <a for="fetch params">request</a>.
+  <li>Let <var>globalObject</var> be <a for="fetch params">request</a>'s
+  <a for=request>client</a>'s <a for="environment settings object">global object</a>.
+  <li>Let <var>timingInfo</var> be <var>fetchParams</var>'s <a for="fetch params">timing info</a>.
+  <li>Let <var>requestedURL</var> be <var>response</var>'s <a for=response>URL list</a>'s first
+  item.
+  <li>Return, if <var>request</var>'s <a for=request>destination</a> is one of the following:
+  <ul>
+    <li><code>"report"</code>
+    <li><code>"document"</code>
+    <li><code>"frame"</code>
```

Comment:
This doesn't cover embed/object/iframe. That doesn't seem intentional.

Reply:
I want to exclude top-level navigations... maybe "document" is enough